### DISCUSSION TITLE
RA fails to start when using systemctl

### DISCUSSION THREAD

**Discussion by odhiambo** [mKeRix/room-assistant#1154] [General]
Hola,

I have installed RA on Pi3B+ running Raspberry OS (Bullseye) using the instructions from the official site:
https://www.room-assistant.io/guide/quickstart-pi.html

When I run RA manually, it runs all well, without any errors.
I then proceed to create the script used by systemd to start/stop RA. Given that I login to my PI as pi3b, so I have the config in /home/pi3b/room-assistant/config.
My /etc/systemd/system/room-assistant.service is modified to reflect that path:
<CUT>
[Unit]
Description=room-assistant service

[Service]
Type=notify
ExecStart=/usr/bin/room-assistant
WorkingDirectory=/home/pi3b/room-assistant
TimeoutStartSec=120
TimeoutStopSec=30
Restart=always
RestartSec=10
WatchdogSec=60
User=pi

[Install]
WantedBy=multi-user.target
</CUT>

But after activating the service, it refuses to start as shown below.
Can anyone figure out what is/could be wrong?

pi3b@pi3B:~/room-assistant $ sudo systemctl enable room-assistant.service
Created symlink /etc/systemd/system/multi-user.target.wants/room-assistant.service → /etc/systemd/system/room-assistant.service.
pi3b@pi3B:~/room-assistant $ sudo systemctl start room-assistant.service
Job for room-assistant.service failed because the control process exited with error code.
See "systemctl status room-assistant.service" and "journalctl -xe" for details.

pi3b@pi3B:~/room-assistant $ sudo systemctl status room-assistant
● room-assistant.service - room-assistant service
     Loaded: loaded (/etc/systemd/system/room-assistant.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Mon 2022-04-18 12:57:28 EAT; 6s ago
    Process: 6895 ExecStart=/usr/bin/room-assistant (code=exited, status=217/USER)
   Main PID: 6895 (code=exited, status=217/USER)
        CPU: 10ms

pi3b@pi3B:~/room-assistant $ journalctl -xe

Apr 18 12:45:51 pi3B systemd[1]: room-assistant.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ The unit room-assistant.service has entered the 'failed' state with result 'exit-code'.
Apr 18 12:45:51 pi3B systemd[1]: Failed to start room-assistant service.
░░ Subject: A start job for unit room-assistant.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ A start job for unit room-assistant.service has finished with a failure.
░░
░░ The job identifier is 5813 and the job result is failed.
Apr 18 12:45:51 pi3B sudo[6803]: pam_unix(sudo:session): session closed for user root
Apr 18 12:46:01 pi3B systemd[1]: room-assistant.service: Scheduled restart job, restart counter is at 65.
░░ Subject: Automatic restarting of a unit has been scheduled
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ Automatic restarting of the unit room-assistant.service has been scheduled, as the result for
░░ the configured Restart= setting for the unit.
Apr 18 12:46:01 pi3B systemd[1]: Stopped room-assistant service.
░░ Subject: A stop job for unit room-assistant.service has finished
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ A stop job for unit room-assistant.service has finished.
░░
░░ The job identifier is 5951 and the job result is done.
Apr 18 12:46:01 pi3B systemd[1]: Starting room-assistant service...
░░ Subject: A start job for unit room-assistant.service has begun execution
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ A start job for unit room-assistant.service has begun execution.
░░
░░ The job identifier is 5951.
Apr 18 12:46:01 pi3B systemd[6807]: room-assistant.service: Failed to determine user credentials: No such process
Apr 18 12:46:01 pi3B systemd[6807]: room-assistant.service: Failed at step USER spawning /usr/bin/room-assistant: No such process
░░ Subject: Process /usr/bin/room-assistant could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ The process /usr/bin/room-assistant could not be executed and failed.
░░
░░ The error number returned by this process is ERRNO.
Apr 18 12:46:01 pi3B systemd[1]: room-assistant.service: Main process exited, code=exited, status=217/USER
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ An ExecStart= process belonging to unit room-assistant.service has exited.
░░
░░ The process' exit code is 'exited' and its exit status is 217.
Apr 18 12:46:01 pi3B systemd[1]: room-assistant.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ The unit room-assistant.service has entered the 'failed' state with result 'exit-code'.
Apr 18 12:46:01 pi3B systemd[1]: Failed to start room-assistant service.
░░ Subject: A start job for unit room-assistant.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ A start job for unit room-assistant.service has finished with a failure.
░░
░░ The job identifier is 5951 and the job result is failed.

**Comment by odhiambo** [comment]
> Hola,
> 
> I have installed RA on Pi3B+ running Raspberry OS (Bullseye) using the instructions from the official site: https://www.room-assistant.io/guide/quickstart-pi.html
> 
> When I run RA manually, it runs all well, without any errors. I then proceed to create the script used by systemd to start/stop RA. Given that I login to my PI as pi3b, so I have the config in /home/pi3b/room-assistant/config. My /etc/systemd/system/room-assistant.service is modified to reflect that path: [Unit] Description=room-assistant service
> 
> [Service] Type=notify ExecStart=/usr/bin/room-assistant WorkingDirectory=/home/pi3b/room-assistant TimeoutStartSec=120 TimeoutStopSec=30 Restart=always RestartSec=10 WatchdogSec=60 User=pi
> 

Answering my stupid self:
I should have completely read the script params and edited the User=pi to reflect the user account I am using. Slaps self :)

**Reply by HiTechTLS** [reply]
Thanks for this, also an idiot
